kimchi/src/circuits/gate.rs

```diff
@@ -217,7 +220,11 @@ pub enum GateLookupTable {
     Xor,
 }
 
-pub type LookupTable<F> = Vec<Vec<F>>;
+/// A table of values that can be used for a lookup, along with the ID for the table.
+pub struct LookupTable<F> {
+    pub id: i32,
+    pub data: Vec<Vec<F>>,
+}
```

Comment:
I agree, it would be better to manage this automatically wherever we can. Although I haven't built anything in this direction, that was part of the motivation for 'negative' table IDs, so that snarky and other library users can allocate runtime tables without concern over how we might change this automatic allocation.